Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -15,7 +20,19 @@ class APIClient {
     /// `APIClient` uses this object to decode the results of network requests.
     let decoder: RequestDecoder
     
+    /// Used for reobtating tokens when they expire and API client receives token expiration error
+    let tokenRefresher: (ClientError, @escaping () -> Void) -> Void
+    
     let cdnClient: CDNClient
+
+    /// Used for syncrhonizing access to requestsQueue
+    private let requestsAccessQueue = DispatchQueue(label: "io.getstream.requests")
+    
+    /// Stores request failed with token expired error for retrying them later
+    private var requestsQueue = [RequestsQueueItem]()
```

Comment:
not sure if we reuse the same instance of APIClient if you login/logout (connectUser/disconnect), if we do we need to make sure that state is reset